build/configure.js

```diff
@@ -0,0 +1,294 @@
+'use strict';
+// Generate and execute makefile
+
+var utils = require('utils');
+
+var PRE_APP_BUILD_STEPS = [
+  'clean-stage-app',
+  'svoperapps',
+  'webapp-manifests',
+  'contacts-import-services',
+  'search-provider',
+  'keyboard-layouts',
+  'preferences',
+  'settings',
+  'webapp-shared',
+  'copy-common-files'
+];
+
+var POST_APP_BUILD_STEPS = [
+  'media-resolution',
+  'post-manifest',
+  'multilocale',
+  'copy-build-stage-data',
+  'webapp-optimize',
+  'webapp-zip'
+];
+
+var Makefile = function(filename) {
+  this.tasks = [];
+  this.filename = filename || this.FILENAME;
+  this.env = utils.getEnv('BUILD_CONFIG');
+};
+
+Makefile.prototype = {
+  FILENAME: 'output.mk',
+
+  insertTask: function(type, target, deps, commands) {
+    this.tasks.unshift({
+      type: type,
+      target: target,
+      deps: deps || [],
+      commands: commands || ''
+    });
+  },
+
+  insertDep: function(target, dep) {
+    var task;
+    for (var id in this.tasks) {
+      task = this.tasks[id];
+      if (task.target === target) {
+        break;
+      }
+    }
+    if (!task) {
+      this.insertTask('phony', target, [dep]);
+    } else if (task.deps.indexOf(dep) === -1) {
+      task.deps.push(dep);
+    }
+  },
+
+  genMakefile: function() {
+    var result = '';
+    var outputFile = utils.getFile(this.filename);
+    this.tasks.forEach(function(task) {
+      if (task.type === 'include') {
+        result += task.target + '\n\n';
+        return;
+      }
+      if (task.type === 'phony') {
+        result += '.PHONY: ' + task.target + '\n';
+      }
+      var depsContent = '';
+        task.deps.forEach(function(dep) {
+          depsContent += ' ' + dep;
+        }, this);
+        result += '' + task.target + ':' + depsContent +
+          '\n\t' + task.commands + '\n\n';
+
+    }, this);
+    utils.writeContent(outputFile, result);
+  },
+
+  executeMakefile: function(target) {
+    var envArray = [target];
+    var envs = {};
+    try {
+      envs = JSON.parse(this.env);
+    } catch (e) {}
+    for (var env in envs) {
+      envArray.push(env + '=' + envs[env]);
+    }
+    envArray.push('-j8');
```

Comment:
I haven't found a way to extract this options ...
http://stackoverflow.com/questions/5303553/gnu-make-extracting-argument-to-j-within-makefile
So I set it as default to run with -j8, or we can detect options.P to decide when to launch parallel.